Mathlib/Analysis/Complex/IntegerCompl.lean

```diff
@@ -0,0 +1,66 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.Analysis.Complex.UpperHalfPlane.Basic
+
+/-!
+# Integer Complement
+
+We define the complement of the integers in the complex plane and give some basic lemmas about it.
+We also show that the upper half plane embeds into the integer complement.
+
+-/
+
+open UpperHalfPlane
+
+/--The complement of the integers in `ℂ`. -/
+def ComplexIntegerComplement := {z : ℂ // ¬ ∃ (n : ℤ), z = ↑n}
+
+namespace ComplexIntegerComplement
+
+local notation "ℂ_ℤ " => ComplexIntegerComplement
+
+noncomputable instance : UniformSpace ℂ_ℤ  :=  instUniformSpaceSubtype
+
+theorem Complex.closedEmbedding_coe_complex : ClosedEmbedding ((↑) : ℤ → ℂ) := by
+  apply Metric.closedEmbedding_of_pairwise_le_dist zero_lt_one
+  convert Int.pairwise_one_le_dist
+  simp_rw [dist_eq_norm]
+  norm_cast
+  rw [Int.norm_eq_abs]
+  exact Int.cast_abs
+
+lemma ℂ_ℤ_Isclosed : IsClosed (((↑) : ℤ → ℂ) '' ⊤) := by
```

Comment:
Please don't use top for sets, but also it seems like this should use `Set.range` directly. Also, this should move to `Mathlib.Analysis.Complex.Basic` as `Complex.isClosed_range_intCast`